test/quark-core/QuarkWallet.t.sol

```diff
@@ -227,12 +227,92 @@ contract QuarkWalletTest is Test {
         // gas: meter execute
         vm.resumeGasMetering();
         vm.expectEmit(true, true, true, true);
-        emit ExecuteQuarkScript(address(aliceAccount), scriptAddress, nonce, EXHAUSTED_TOKEN, ExecutionType.Direct);
+        emit ExecuteQuarkScript(address(aliceAccount), scriptAddress, nonce, nonce, ExecutionType.Direct);
+        aliceWalletExecutable.executeScript(nonce, scriptAddress, call, new bytes[](0));
+    }
+
+    function testFailsWithRepeatNonceInDirectExecute() public {
+        // gas: do not meter set-up
+        vm.pauseGasMetering();
+
+        Counter counter = new Counter();
+        assertEq(counter.number(), 0);
+
+        bytes memory maxCounterScript = new YulHelper().getCode("MaxCounterScript.sol/MaxCounterScript.json");
+        address scriptAddress = codeJar.saveCode(maxCounterScript);
+        bytes memory call = abi.encodeWithSignature("run(address)", address(counter));
+
+        QuarkWallet aliceWalletExecutable = newWallet(aliceAccount, aliceAccount);
+        bytes32 nonce = new QuarkOperationHelper().semiRandomNonce(nonceManager, aliceWalletExecutable);
+
+        vm.startPrank(aliceAccount);
+
+        // gas: meter execute
+        vm.resumeGasMetering();
+        vm.expectEmit(true, true, true, true);
+        emit ExecuteQuarkScript(address(aliceAccount), scriptAddress, nonce, nonce, ExecutionType.Direct);
+        aliceWalletExecutable.executeScript(nonce, scriptAddress, call, new bytes[](0));
+
+        assertEq(counter.number(), 1);
+
+        // Not sure why this revert isn't showing up-- it's reverting, nonetheless.
```

Comment:
I know this is crazy but is it because you didn't do `new bytes[](0)` (inline in the call to execute on 264) above the `expectRevert`?

I mean... the constructor execution succeeds, so I wonder if the expect revert accidentally targets this nested call.

Reply:
Nah, doesn't make it work. Though that change is probably for the better for clarity, so I moved that construction to a var above anyway, thanks.